### PR TITLE
Use pseudo-code for the signature creation and verification examples

### PR DESCRIPTION
**Pull request by glpatcern**
This is in response to my own remark that pseudo-code is more appropriate in specs. I hope I got the process right!

### CODE REVIEW

**Review by MahdiBaghbani — APPROVED** [review]
It is more general than the PHP (looks like Python now 😄) 
LGTM

**Review by mickenordin — APPROVED** [review]

**Comment by glpatcern** [comment]
> It is more general than the PHP (looks like Python now 😄) LGTM

Surely I got inspired by Python :-D but hopefully the readability got improved

File: README.md [hunk 1]
```diff
@@ -404,73 +404,71 @@ Here is an example of headers needed to sign a request.
...
+    '(request-target)': 'post /path',
+    'content-length': length_of(payload),
+    'date': current_gmt_datetime(),  # Use a function to get the current GMT date as 'D, d M Y H:i:s T'
+    'digest': 'SHA-256=' + base64_encode(hash('sha256', utf8_encode(payload))),
+    'host': 'recipient-hostname',
+}
+
+signed = ssl_sign(concatenate_with_newlines(headers), private_key, 'sha256')
+signature = {
+    'keyId': 'https://sender.org/ocm',  # The sending server's keyId as advertised at the discovery endpoint
```
Comment (glpatcern) [inline README.md]: Reviewing myself, I modified this to clarify that the keyId is per sending server (and not per user)

**Review by glpatcern — COMMENTED** [review]
I edited again the example, @michielbdejong a final pass?

File: README.md [hunk 2]
```diff
@@ -404,73 +404,71 @@ Here is an example of headers needed to sign a request.
...
-    $headers['Signature'] = implode(',', $signature);
+This is a pseudo-code example for generating the `Signature` header for outgoing requests:
+
+```code
+headers = {
+    '(request-target)': 'post /path',
+    'content-length': length_of(payload),
+    'date': current_gmt_datetime(),  # Use a function to get the current GMT date as 'D, d M Y H:i:s T'
+    'digest': 'SHA-256=' + base64_encode(hash('sha256', utf8_encode(payload))),
+    'host': 'recipient-hostname',
```
Comment (michielbdejong) [inline README.md]:
```suggestion
    'host': 'recipient-fqdn',
```

File: README.md (hunk 1 above)
Comment (michielbdejong) [inline README.md]:
```suggestion
    'keyId': 'sender-fqdn',  # The sending server's FQDN; find its public key through OCM API discovery
```

File: README.md [hunk 3]
```diff
@@ -404,73 +404,71 @@ Here is an example of headers needed to sign a request.
...
-    if (openssl_verify(implode("\n", $clear), $signed, $publicKey, 'sha256') !== 1) {
-        throw new InvalidSignatureException('signature issue');
-    }
+```code
+clear = {
+    '(request-target)': 'post /path',
+    'content-length': length_of(payload),
+    'date': 'Mon, 08 Jul 2024 14:16:20 GMT',  # The date used in the verification process
+    'digest': 'SHA-256=' + base64_encode(hash('sha256', utf8_encode(payload))),  # Recompute the digest for verification
+    'host': localhost(),
```
Comment (michielbdejong) [inline README.md]:
```suggestion
    'host': 'example.com',
```

**Review by michielbdejong — CHANGES_REQUESTED** [review]
Great stuff - only remark, let's use FQDNs here too instead of URLs, for consistency.

**Comment by glpatcern** [comment]
> Great stuff - only remark, let's use FQDNs here too instead of URLs, for consistency.

Good point, just committed your suggestions (and a trivial variant). Good to go?

**Review by michielbdejong — APPROVED** [review]

**Pull request merged**